umm-spec-lib/test/cmr/umm_spec/test/xml_to_umm_mappings/iso_shared/use_constraints.clj

```diff
@@ -0,0 +1,308 @@
+                       <gmd:resourceConstraints>
+                         <gmd:MD_LegalConstraints>
+                           <gmd:otherConstraints>
+                             <gco:CharacterString>Access Constraints Description:  description</gco:CharacterString>
+                           </gmd:otherConstraints>
+                         </gmd:MD_LegalConstraints>
+                       </gmd:resourceConstraints>
+                       <gmd:aggregationInfo>")
+      smap-xpath
+      uc/other-constraints-xpath
+      uc/new-access-desc
+      "  description")))
+
+(deftest parse-access-constraints-test
+  (testing "Parsing access constraints"
+
+    (are3 [test-iso xpath sanitize? expected]
+      (is (= expected (uc/parse-access-constraints test-iso xpath sanitize?)))
+
+      "Test no constraints for access constraints value MENDS using old key"
+      base-iso iso-xpath true nil
+
+      "Test Access Constraints Value constraints without space for access constraints value.
+       Description will be Not provided since sanitize? is true."
+      (string/replace base-iso
+                      #"<\/gmd:descriptiveKeywords>[ *|\n|\r]*<gmd:aggregationInfo>"
+                      "</gmd:descriptiveKeywords>
+                       <gmd:resourceConstraints>
+                         <gmd:MD_LegalConstraints>
+                           <gmd:otherConstraints>
+                             <gco:CharacterString>Access Constraints Value:15</gco:CharacterString>
+                           </gmd:otherConstraints>
+                         </gmd:MD_LegalConstraints>
+                       </gmd:resourceConstraints>
+                       <gmd:aggregationInfo>")
+      iso-xpath
+      true
+      {:Description "Not provided" :Value 15.0}
+
+      "Test Access Constraints Value constraints using smap for Restriction Flag.
+       Restriction Flag will be used and the other will be ignored.
+       Description will be nil since sanitize? is false."
+      (string/replace base-smap
+                      #"<\/gmd:descriptiveKeywords>[ *|\n|\r]*<gmd:aggregationInfo>"
+                      "</gmd:descriptiveKeywords>
+                       <gmd:resourceConstraints>
+                         <gmd:MD_LegalConstraints>
+                           <gmd:otherConstraints>
+                             <gco:CharacterString>Restriction Flag: 15</gco:CharacterString>
+                           </gmd:otherConstraints>
+                           <gmd:otherConstraints>
+                             <gco:CharacterString>Access Constraints Value: 20</gco:CharacterString>
+                           </gmd:otherConstraints>
+                         </gmd:MD_LegalConstraints>
+                       </gmd:resourceConstraints>
+                       <gmd:aggregationInfo>")
+      smap-xpath
+      false
+      {:Value 15.0 :Description nil}
+
+      "Test Access Constraints Value constraints using smap for Restriction Flag.
```

Comment:
For completeness, we should have a test where the AC value is not a double.

Reply:
I do, the next test case is where the value is "abc" which is not a double. All of the other test cases, the value is actually an integer, but the code translates the number to a double.